internal/services/videoanalyzer/video_analyzer_edge_module_resource_test.go

```diff
@@ -0,0 +1,146 @@
+package videoanalyzer_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/videoanalyzer/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type VideoAnalyzerEdgeModuleResource struct {
+}
+
+func TestAccVideoAnalyzerEdgeModule_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_video_analyzer_edge_module", "test")
+	r := VideoAnalyzerEdgeModuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check:  acceptance.ComposeAggregateTestCheckFunc(),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccVideoAnalyzerEdgeModule_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_video_analyzer_edge_module", "test")
+	r := VideoAnalyzerEdgeModuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check:  acceptance.ComposeAggregateTestCheckFunc(),
```

Comment:
can we make this:

```suggestion
			Check:  acceptance.ComposeAggregateTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
```

Reply:
I've made this into a check of the `video_analyzer_name` name now

Reply:
the values of the `video_analyzer_name` field are checked in the `import` below (which validates we can import this from the API) - so it'd be better to check the resource exists in the Azure API (e.g. to check we're not tracking a resource which has been provisioned with the wrong name or something) than the specific value here:

```suggestion
 				check.That(data.ResourceName).ExistsInAzure(r),
```